newrelic/hooks/mlmodel_openai.py

```diff
@@ -54,46 +53,75 @@ def wrap_embedding_create(wrapped, instance, args, kwargs):
     if not transaction:
         return wrapped(*args, **kwargs)
 
+    # Framework metric also used for entity tagging in the UI
     transaction.add_ml_model_info("OpenAI", OPENAI_VERSION)
 
+    # Obtain attributes to be stored on embedding events regardless of whether we hit an error
+    request_args = kwargs
+    embedding_id = str(uuid.uuid4())
+
+    # Get API key without using the response so we can store it before the response is returned in case of errors
+    api_key = getattr(openai, "api_key", None)
+    api_key_last_four_digits = f"sk-{api_key[-4:]}" if api_key else ""
+
+    # Get trace information
+    available_metadata = get_trace_linking_metadata()
+    span_id = available_metadata.get("span.id", "")
+    trace_id = available_metadata.get("trace.id", "")
+
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+
+    # Shared attributes that will be included on all embedding events (error or not)
+    base_embedding_dict = {
+        "embedding_id": embedding_id,
+        "appName": settings.app_name,
+        "api_key_last_four_digits": api_key_last_four_digits,
+        "span_id": span_id,
+        "trace_id": trace_id,
+        "transaction_id": transaction.guid,
+        "input": request_args.get("input", ""),
+        "request.model": request_args.get("model") or request_args.get("engine") or "",
+        "vendor": "openAI",
+        "ingest_source": "Python",
+    }
+
     ft_name = callable_name(wrapped)
     with FunctionTrace(ft_name) as ft:
         try:
             response = wrapped(*args, **kwargs)
         except Exception as exc:
-            error_attributes = openai_error_attributes(exc, kwargs)
-            exc._nr_message = error_attributes.pop("error.message")
+            notice_error_attributes = {
+                "http.statusCode": getattr(exc, "http_status", ""),
+                "error.message": getattr(exc, "_message", ""),
+                "error.code": getattr(getattr(exc, "error", ""), "code", ""),
+                "error.param": getattr(exc, "param", ""),
+            }
+            exc._nr_message = notice_error_attributes.pop("error.message")
             ft.notice_error(
-                attributes=error_attributes,
+                attributes=notice_error_attributes,
             )
+            # Gather attributes to add to embedding summary event in error context
+            error_summary_attributes = {
+                "response.organization": getattr(exc, "organization", ""),
```

Comment:
Note to reviewer to chime in: It feels strange to call this `response.organization` even though we didn't get in from the response in the error case. We could change this to just be `organization` for all summary events.

Reply:
It is still coming from a response in a way it's just the error response but I see your point. If we change it though, the UI will have to change with it so for now I'd just leave it. It's definitely worth having the discussion on these names before GA though since it will be really hard to change them after that.